lua/cfc_time/server/time_tracking.lua

```diff
@@ -56,30 +60,29 @@ function ctime:updateTimes()
     local now = getNow()
     local timeDelta = now - self.lastUpdate
 
-    for steamID, data in pairs( self.sessions ) do
-        local isValid = true
+    local sessionIDs = self.sessionIDs
+    local totalTimes = self.totalTimes
 
+    for steamID, data in pairs( self.sessions ) do
         local joined = data.joined
         local departed = data.departed
 
+        -- If they've departed and we already saved the session to the DB, we can skip and untrack
         if departed and departed < self.lastUpdate then
             self:untrackPlayer( steamID )
-            isValid = false
-        end
-
-        local sessionTime = ( departed or now ) - joined
-        if sessionTime <= 0 then
-            isValid = false
-        end
-
-        if isValid then
+        else
+            local sessionTime = ( departed or now ) - joined
             data.duration = sessionTime
 
-            local sessionID = self.sessionIDs[steamID]
+            local sessionID = sessionIDs[steamID]
             batch[sessionID] = data
 
-            local newTotal = self.totalTimes[steamID] + timeDelta
-            self.totalTimes[steamID] = newTotal
+            -- Players may not have a total time if they weren't set up successfully
+            local currentTotal = totalTimes[steamID]
+            if currentTotal then
+                local newTotal = currentTotal + timeDelta
+                totalTimes[steamID] = newTotal
+            end
```

Comment:
Diff messed this up.

Main changes:
- Use local table lookups
- No more `isValid` variable. Instead, prefer nested if/elses to make the flow clearer
- The player's `totalTime` is not updated if they don't have one. This is necessary because players who DC during the database operation will not have a `totalTime` set. This actually fixes the NULL entity error as well.